test/test_utils/ec2.py

```diff
@@ -435,42 +437,120 @@ def execute_ec2_inference_test(connection, ecr_uri, test_cmd, region=DEFAULT_REG
+        f"{docker_cmd} run --user root "
+        f"-e LOG_FILE={os.path.join(os.sep, 'test', 'benchmark', 'logs', log_name)} "
+        f"-v {container_test_local_dir}:{os.path.join(os.sep, 'test')} {ecr_uri} "
         f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}"
     )
+    ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location, data_source,
+                                                                 threshold, post_process, log_name)
 
 
-def execute_ec2_inference_performance_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION):
+def execute_ec2_inference_performance_test(connection, ecr_uri, test_cmd, region=DEFAULT_REGION,
+                                           post_process=None, data_source="", threshold=None):
     docker_cmd = "nvidia-docker" if "gpu" in ecr_uri else "docker"
     container_test_local_dir = os.path.join("$HOME", "container_tests")
-
+    timestamp = time.strftime('%Y-%m-%d-%H-%M-%S')
+    log_name = f"{data_source}_results_{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}_{timestamp}.txt"
     # Make sure we are logged into ECR so we can pull the image
     connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
-
     connection.run(f"{docker_cmd} pull -q {ecr_uri}")
 
     # Run training command, display benchmark results to console
     repo_name, image_tag = ecr_uri.split("/")[-1].split(":")
     container_name = f"{repo_name}-performance-{image_tag}-ec2"
     connection.run(
         f"{docker_cmd} run -d --name {container_name} "
-        f"-e COMMIT_INFO={os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')} "
+        f"-e LOG_FILE={os.path.join(os.sep, 'test', 'benchmark', 'logs', log_name)} "
         f"-v {container_test_local_dir}:{os.path.join(os.sep, 'test')} {ecr_uri}"
     )
     try:
-        connection.run(f"{docker_cmd} exec --user root {container_name} " f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}")
+        connection.run(
+            f"{docker_cmd} exec --user root {container_name} " f"{os.path.join(os.sep, 'bin', 'bash')} -c {test_cmd}")
     except Exception as e:
         raise Exception("Failed to exec benchmark command.\n", e)
     finally:
         connection.run(f"docker rm -f {container_name}")
+    log_location = os.path.join(container_test_local_dir, "benchmark", "logs", log_name)
+    ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location, data_source,
+                                                                 threshold, post_process, log_name)
+
+
+def ec2_performance_upload_result_to_s3_and_validate_performance(connection, ecr_uri, log_location,
+                                                                 data_source, threshold, post_process, log_name):
+    framework = "tensorflow" if "tensorflow" in ecr_uri else "mxnet" if "mxnet" in ecr_uri else "pytorch"
+    framework_version = re.search(r"[0-9]+(\.\d+){2}", ecr_uri).group()
```

Comment:
`r"\d"` is exactly the same as `r"[0-9]"`. Please use one or the other.

Reply:
'\d' is more concise. will turn to use this one